crates/core/tedge/src/error.rs

```diff
@@ -44,4 +45,14 @@ pub enum TEdgeError {
 
     #[error(transparent)]
     FromC8yEndPointConfigError(#[from] c8y_api::http_proxy::C8yEndPointConfigError),
+
+    #[error(
+        r#"The given device ID '{input_id}' doesn't match the one in the config '{config_id}'.
+    Run `tedge config unset {writable_key}` first to unset the device ID."#
+    )]
+    UnmatchedDeviceId {
```

Comment:
```suggestion
    MismatchedDeviceId {
```

Reply:
fixed in https://github.com/thin-edge/thin-edge.io/pull/3326/commits/ed4043332d41a517b08922c90f7a3e55ce4e8c7c